### PR TITLE
fix(image): remove all image when image source changes

### LINKED ISSUE #8607: Using `Image` in a list briefly shows previously seen image due to recycling
### Current behavior

Consider the following template:

```xaml
<DataTemplate x:DataType="octokit:User">
    <Grid Padding="8" ColumnSpacing="16">
        <Grid.RowDefinitions>
            <RowDefinition Height="Auto" />
            <RowDefinition Height="Auto" />
        </Grid.RowDefinitions>
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="Auto" />
            <ColumnDefinition Width="*" />
        </Grid.ColumnDefinitions>
        <Image Source="{x:Bind AvatarUrl}" Width="64" Height="64" Grid.RowSpan="2" />
        <TextBlock Style="{ThemeResource TitleTextBlockStyle}" Text="{x:Bind Login}" Grid.Column="1" />
        <HyperlinkButton Content="{x:Bind HtmlUrl}" NavigateUri="{x:Bind HtmlUrl}" Grid.Row="1" Grid.Column="1" Padding="0" />
    </Grid>
</DataTemplate>
```

When data binding a large list, the items get recycled and you can briefly see the new items show images from some previous entries before the actual image loads due to the fact that the template is recycled.

### Expected behavior

Should not happen - images should show up as blank by default.

Seen in Skia & WASM, but may happen on other platforms as well

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

-

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

4.2

### Affected platforms

WebAssembly, Skia (GTK on Linux/macOS/Windows)

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### LINKED ISSUE #15861: Icons in Listview slow to Update
I have a strange issue, which I'm not sure if it's something I'm doing, or a bug. I have a page with a ListView, which then has a WrapPanel from the community toolkit. I'm displaying Toolkit Cards in it with an icon, two lines of text. It looks like this, works great:
image

If I click on one of the cards, I'm navigating to a sub menu, which is essentially the same view, but with a new instance of a view model (that has a new set of menu items, in a `ObserverableCollection`). The issue is that for a split second (which is more obvious on a RPi that is slower to process) the icons from the first list of menu items are used on the new list. It looks like this:
image

But the next "frame" shows the correct icons:
image

I can't figure out why this happens. I've traced the code, I get my new instance of the VM, I set the list of menu items, it never has the old icons with the new menu items. It's like the icons are cached or something. Is there a workaround where I can pause screen refresh until everything is updated, or another way to solve it? It looks strange when the icons changed after the menu is updated.

_Originally posted by @HakanL in https://github.com/unoplatform/uno/discussions/14047_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #8607 closes #15861

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Image/Image.skia.cs [hunk 1]
```diff
@@ -28,7 +28,14 @@ public Image()
...
 		partial void OnSourceChanged(ImageSource newValue, bool forceReload)
 		{
+			// We clear the old image first. We do NOT wait for the new image to load
+			// for it to replace the old one. This is what happens on WinUI.
 			_sourceDisposable.Disposable = null;
+			_lastMeasuredSize = default;
+			_imageSprite.Brush = null;
+			_currentSurface = null;
+			_pendingImageData = new();
+			InvalidateMeasure();
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Image/Image.skia.cs:38]: @ramezgerges Can you try on winui to see if, when using a very slow URL, the image disapears immediately (if there already was an image displayed)? I recall some other types of behaviors around this on iOS/Android.

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/Image/Image.skia.cs:38]: Yes, WinUI will immediately remove the old image (and its allocated area) and then acquire the image area back once the new image loads. It looks pretty jarring.

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/Image/Image.skia.cs:38]: @ramezgerges I vaguely recall issues around databinding in item templates, particularly around databinding the source. Is this the scenario that you tested ?

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/Image/Image.skia.cs:38]: I checked with a binding and it's behaving the same as you described.

**Comment by jeromelaban** [comment]
@ramezgerges Looking at the test failures, it seems that the changes are causing a memory leak.

**Comment by ramezgerges** [comment]
I've tested locally and it seems that the failing test is flaky and it's also flaky on master.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**